### ISSUE TITLE
<12/12/23><NY><Android><Reallocation> If a driver accepts & then cancels the ride, that driver is not receiving RR when the user searches by adding a tip, either on the ride search screen or the 'Try again' screen.<parent id 1980>

### ISSUE THREAD

**Issue by Gowthami1309** [nammayatri/nammayatri#4715]
#1980

**Describe the bug**
If a driver accepts and then cancels the ride, that driver is not receiving ride requests when the user searches by adding a tip, either on the ride search screen or the 'Try again' screen.

**To Reproduce**

1. Make 2 drivers and 1 User
2. Search RR from User side in "auto assign" mode
3. complete booking from D1
4. cancel the Ride from D1
5. Search with tip from ride search screen or from try again screen
6. And observe D1 side 

**Expected behavior**
If a driver accepts and then cancels the ride, that driver is not receiving ride requests when the user searches by adding a tip, either on the ride search screen or the 'Try again' screen.

**Attachment**

**App (please complete the following information):**

Device: All
Version All

**Comment by 123ajay1994** [comment]
@ashishsingh101 Any update on it? 
here we are talking about blocked driver if they can receive reallocated RR or not if tip has been included. 

cc: @SrinivasNaikB

**Comment by ashishsingh101** [comment]
@123ajay1994 if the driver cancels the ride then that driver is blacklisted for that searchReq and he should not get the same ride again even with the Tip, @SrinivasNaikB can you comment on this ?

**Comment by 123ajay1994** [comment]
@ashishsingh101 Agree, But basically in general, we do share RR with tip even when the RR count get exhausted at BPP side(in Initial Search_Try. So I have raised this one!!
